### PR TITLE
[monotouch-test] Don't assert any specific value for the ReusedConnection value in UrlSessionTaskTransactionMetricsTest. Fixes #xamarin/maccore@2281.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Fixes https://github.com/xamarin/maccore/issues/2281.

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
✅ [Jenkins job](https://jenkins.internalx.com/job/macios/job/PR-9660/1/display/redirect) (on internal Jenkins) succeeded

✅ [Provisioning succeeded](https://jenkins.internalx.com/job/macios/job/PR-9660/1//console)
✅ [Build succeeded](https://jenkins.internalx.com/job/macios/job/PR-9660/1//console)
✅ Packages built successfully
<details><summary>View packages</summary>

* [xamarin.ios-13.23.0.201.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/xamarin.ios-13.23.0.201.pkg)
* [xamarin.mac-6.23.0.201.pkg (Not notarized)](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/xamarin.mac-6.23.0.201.pkg)
* [Microsoft.iOS.Ref.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Ref.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.iOS.Runtime.ios-arm.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Runtime.ios-arm.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.iOS.Runtime.ios-arm64.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Runtime.ios-arm64.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.iOS.Runtime.ios-x64.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Runtime.ios-x64.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.iOS.Runtime.ios-x86.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Runtime.ios-x86.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.iOS.Sdk.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.iOS.Sdk.13.6.100-ci.pr.gh9660.373+sha.ac96f06e3.nupkg)
* [Microsoft.macOS.Ref.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.macOS.Ref.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.macOS.Runtime.osx-x64.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.macOS.Sdk.10.15.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.tvOS.Ref.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.tvOS.Ref.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.tvOS.Runtime.tvos-arm64.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.tvOS.Runtime.tvos-x64.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.tvOS.Sdk.13.4.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.watchOS.Ref.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.watchOS.Ref.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.watchOS.Runtime.watchos-arm.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.watchOS.Runtime.watchos-x86.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)
* [Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg](https://bosstoragemirror.blob.core.windows.net/wrench/jenkins/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/package/Microsoft.watchOS.Sdk.6.2.100-ci.pr.gh9660.731+sha.ac96f06e3.nupkg)

</details>

✅ [API Diff (from stable)](http://xamarin-storage/jenkins/xamarin-macios/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/jenkins-results/api-diff/index.html)
✅ [API Diff (from PR only)](http://xamarin-storage/jenkins/xamarin-macios/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/jenkins-results/apicomparison/api-diff.html) (no change)
✅ [Generator Diff](http://xamarin-storage/jenkins/xamarin-macios/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/jenkins-results/generator-diff/index.html) (no change)
✅ [Test run succeeded](http://xamarin-storage/jenkins/xamarin-macios/PR-9660/ac96f06e3b9786e0dfae4af290b6bac155f95c74/1/jenkins-results/tests/index.html)

**Review by whitneyschmidt — APPROVED** [review]

File: tests/monotouch-test/Foundation/UrlSessionTaskTransactionMetricsTest.cs [hunk 1]
```diff
@@ -58,12 +58,7 @@ public void Properties ()
 					Assert.Null (sttm.ResponseEndDate, "ResponseEndDate");
 					Assert.Null (sttm.ResponseStartDate, "ResponseStartDate");
 				}
-#if __IOS__
-				if (TestRuntime.CheckXcodeVersion (11, 6)) // Returns true in iOS 13.6
-					Assert.True (sttm.ReusedConnection, "ReusedConnection");
-				else
-#endif
-					Assert.False (sttm.ReusedConnection, "ReusedConnection");
+				Assert.That (sttm.ReusedConnection, Is.EqualTo (true).Or.EqualTo (false), "ReusedConnection");
```
Comment (tj-devel709) [inline tests/monotouch-test/Foundation/UrlSessionTaskTransactionMetricsTest.cs:61]: Why do we need to check this value if it passes regardless of true or false? Is it a nullable bool perhaps and we want to make sure its not null?

Comment (mandel-macaque) [reply tests/monotouch-test/Foundation/UrlSessionTaskTransactionMetricsTest.cs:61]: The return type is not a nullable, so it is not a bool?. Which means that we need to assert that the property works, but we do not care about the value, and we cant check against null. How do we do that with a value type and ignore the value? We could have done this Assert.True (true || sttm.ReusedConnection, "ReusedConnection") but I think that the way @rolfbjarne wrote it is more explicit.

**Review by mandel-macaque — APPROVED** [review]

File: tests/monotouch-test/Foundation/UrlSessionTaskTransactionMetricsTest.cs (hunk 1 above)
Comment (tj-devel709) [reply tests/monotouch-test/Foundation/UrlSessionTaskTransactionMetricsTest.cs:61]: Thanks!

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**